### ISSUE TITLE
IOS Notification Not Receiving less then 10 Versions

### ISSUE THREAD

**Issue by Haideralee** [fechanique/cordova-plugin-fcm#247]
I am testing with multiple **ios** devices and i failed to receive any notification that devices which version is less then 10 ,in just iphone 4s (IOS 9.4) i just received notification when app is not opened or app is on background but **FCMPlugin.onNotification** never work.it is so wired.

**Comment by yoprogramo** [comment]
@Haideralee : I can confirm it. All my devices iOS 9 are not receiving notifications, but iOS 10 yes..

**Comment by Haideralee** [comment]
@yoprogramo yes but i have iphone 4s ios 9.4 it is just receiving background or when app is not open. i confirmed that **FCMPlugin.onNotification** event not fire

**Comment by yoprogramo** [comment]
@Haideralee I have iOS 9.3.5 (ipad mini) and I don't receive notification in background nor app closed.

**Comment by Haideralee** [comment]
@yoprogramo yes it is wired behaviors i mentioned that just iphone 4s (IOS 9.4) receiving notification when app is not active mode and i also 💯 sure that  **FCMPlugin.onNotification** event never fire.

**Comment by arvin8** [comment]
iOS8.3 iPad v1, receives when app closed or in background

**Comment by elewin** [comment]
Same thing here with a an iOS9.3.5 iphone and an iOS 9.32 ipad -- they receive background notifications but not when the app is open. However we have a couple iOS 10 ipads and iphones we test on too that do work in both cases

**Comment by Haideralee** [comment]
@elewin yes i have tested too with IOS 10 devices and it's work perfectly but it's not working on less then 10 IOS version.

**Comment by ferdil** [comment]
There's a coding problem in the file src/ios/AppDelegate+FCMPlugin.m

It uses some #defined macros (specifically 
`#if defined(__IPHONE_10_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_10_0`
This doesn't check the version of iOS running - it checks the version of the SDK being used to compile the code. As such, this excludes the code needed to run on pre-iOS 10 devices (where didReceiveRemoteNotification is required) 

The solution is to remove the #if, #else and #endif in lines 108, 161 and 213 of AppDelegate+FCMPlugin.m
That gets it working on all version of iOS

Ferdi

**Comment by yoprogramo** [comment]
@ferdil tested removing the macros, it works on my iPad with iOS 9.3.5. Can you make a pull request or it is something provisional?

**Comment by Shakti-Sharma** [comment]
Boss Please solve this issue ASAP... it was working well when older version about 1.2 or 1.4. But i found some issues on Android devices where background push is not working. So i was to update it to the current version. 

Kindly please update the revision.
Thanks
Shakti

**Comment by ferdil** [comment]
I'm battling to create a pull request. Maybe @fechanique can do the change ?

**Comment by Haideralee** [comment]
@ferdil @fechanique is not responding. it is so wired. :(

**Comment by yoprogramo** [comment]
I'll try to generate the pull request by myself...
